src/nodewidget.rs

```diff
@@ -0,0 +1,215 @@
+use conrod::{UiId, Ui};
+use conrod::{WidgetUpdate};
+use conrod::CharacterCache;
+
+#[derive(PartialEq, Copy, Clone, Debug)]
+pub enum NodeState {
+    Normal, //expanded, static position; todo: rename to static?
+    NormalCollapsed, // alternative to enum states for collapse, we can use collapse state as: Normal(true) or track a separate collapse state enum
+    Dragging,
+}
+
+fn get_new_state(is_over: bool, prev: NodeState, mouse: Mouse) -> NodeState {
+    use self::NodeState::{Normal,NormalCollapsed, Dragging};
+    //buttonstate is either up or down, if there was a 'released' state we could track this easily
+    match (is_over, prev, (mouse.left,mouse.right)) {
+        (true,  _, (MouseButtonState::Down,_)) => Dragging,
+        (true, Normal, (_,MouseButtonState::Down)) => NormalCollapsed, //this tracks bad because the state flips too quickly, a delay or mouse-release would fix this
+        (true, NormalCollapsed, (_,MouseButtonState::Down)) => Normal,
+        
+        (false, Dragging, (MouseButtonState::Down,_)) => Dragging,
+        
+        (false, Dragging,(MouseButtonState::Up,_)) => Normal,
+        (true, Dragging,(MouseButtonState::Up,_)) => Normal,
+        _ => prev,
+    }
+}
+
+impl NodeState {
+    /// Alter the widget color depending on the state.
+    fn color(&self, color: Color) -> Color {
+        match *self {
+            NodeState::Normal => color,
+            NodeState::NormalCollapsed => color.complement(),
+            NodeState::Dragging => color.highlighted(),
+        }
+    }
+}
+
+impl CustomWidgetState for NodeState {
+    fn init() -> Self { NodeState::Normal }
+    fn matches(&self, _other: &NodeState) -> bool { true }
+}
+
+#[derive(Clone, Debug)]
+pub struct Node<'a> {
+    pos: Position,
+    dim: Dimensions,
+    depth: Depth,
+    maybe_label: Option<&'a str>,
+    style: Style,
+}
+
+#[derive(PartialEq, Clone, Debug)]
+pub struct Style {
+    pub maybe_color: Option<Color>,
+    pub maybe_frame: Option<f64>,
+    pub maybe_frame_color: Option<Color>,
+    pub maybe_label_color: Option<Color>,
+    pub maybe_label_font_size: Option<u32>,
+}
+
+impl Style {
+    pub fn new() -> Style {
+        Style {
+            maybe_color: None,
+            maybe_frame: None,
+            maybe_frame_color: None,
+            maybe_label_color: None,
+            maybe_label_font_size: None,
+        }
+    }
+}
+
+impl<'a> Node<'a> {
+    pub fn new() -> Node<'a> {
+        Node {
+            pos: Position::Absolute(0.0, 0.0),
+            dim: [64.0, 64.0],
+            depth: 0.0,
+            maybe_label: None,
+            style: Style::new(),
+        }
+    }
+}
+
+impl<'a> CustomWidget for Node<'a> {
+
+    type State = NodeState;
+
+    fn update<C>(self, state: NodeState, ui_id: UiId, ui: &mut Ui<C, Node<'a>>) -> WidgetUpdate<NodeState>
+        //where C: CharacterCache //I had to comment this out, not sure why it wouldn't compile for me
```

Comment:
I sent a PR to Conrod, I think it was merged today, so if you update it should work again.